contracts/ConverterV2.sol

```diff
@@ -0,0 +1,210 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "./polydex/interfaces/IPolydexERC20.sol";
+import "./polydex/interfaces/IPolydexPair.sol";
+import "./polydex/interfaces/IPolydexFactory.sol";
+import "./polydex/interfaces/IPolydexRouter.sol";
+
+// ConverterV2 is Farm's left hand and kinda a wizard. He can create up CNT from pretty much anything!
+// This contract handles "serving up" rewards for xCNT holders & also burning some by trading tokens collected from fees for CNT.
+// This contract differs in the way from its predecessor by directly using the Polydex Router to swap LP tokens and ERC20 tokens to CNT.
+
+contract ConverterV2 is Ownable, ReentrancyGuard {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    
+
+    IPolydexFactory public factory;
+    IPolydexRouter public router;
+    address public cntStaker;
+    // The CNT TOKEN!
+    IERC20 public cnt;
+    address public wmatic;
+    address public l2Burner;
+    uint16 public burnAllocation;
+    uint16 public stakersAllocation;
+    uint16 public platformFeesAllocation;
+    address public platformAddr;
+
+    event CNTConverted(
+        uint256 stakersAllocated,
+        uint256 burnt,
+        uint256 platformFees
+    );
+
+    modifier ensureNonZeroAddress(address addressToCheck) {
+        require(addressToCheck != address(0), "No zero address");
+        _;
+    }
+
+    constructor(
+        IERC20 _cnt,
+        uint16 _burnAllocation,
+        uint16 _stakersAllocation,
+        uint16 _platformFeesAllocation,
+        address _platformAddr
+    ) {
+        cnt = _cnt;
+        platformAddr = _platformAddr;
+        setAllocation(
+            _burnAllocation,
+            _stakersAllocation,
+            _platformFeesAllocation
+        );
+        //Set to Polydex Contract Addresses deployed on Matic Mainnet
+        router = IPolydexRouter(0xBd13225f0a45BEad8510267B4D6a7c78146Be459);
```

Comment:
Possible take this as constructor arguments

Reply:
Hey thanks for checking out.
We added this directly as asked by @ashwinYardi to have less params in constructor and save gas.